_includes/graphql/optimization.md

```diff
@@ -0,0 +1,177 @@
+# Optimization
+
+## Indexing
+
+To optimize your server/database performance and response time you need to configure some indexes based on your use cases (types of queries that you use into your app).
+Currently indexes manipulations are not supported on the GraphQL API but you can use on the server side the [Parse JS SDK Indexes API](https://docs.parseplatform.org/js/guide/#indexes)
+
+Here a simple example:
+```graphql
+# First we update the user class
+mutation {
+  updateClass(
+    input: {
+      name: "_User"
+      schemaFields: {
+        addStrings: [{ name: "firstname" }, { name: "lastname" }]
+        addDates: [{ name: "birthdate" }]
+        addNumbers: [{ name: "credit" }]
+      }
+    }
+  ) {
+    class {
+      name
+    }
+  }
+}
+```
+```graphql
+# Then lets assume that we have this query in our app
+query ASuperUsedQuery {
+  users(
+    where: {
+      firstname: { in: ["John", "Michael", "Sabrina"] }
+      lastname: { matchesRegex: "^Do" }
+      birthdate: { exists: true }
+      credit: { greaterThanOrEqualTo: 100 }
+    },
+    order: credit_ASC
+  ) {
+    edges {
+      node {
+        id
+        firstname
+      }
+    }
+  }
+}
+```
+```js
+// Add the index to user class
+const addAnIndexToUserClass = () => {
+    // Get current class from database
+    const UserClass = await (new Parse.Schema('_User')).get();
+    UserClass.addIndex('ASuperUsedQueryIndex', {
+        firstname: 1,
+        lastname: 1,
+        birthdate: 1,
+        credit: 1
+    });
+    await UserClass.update()
+    console.log('Index for ASuperUsedQuery created')
+}
+```
+And thats all, your query is optimized at 100% and you should notice blazing fast response even on 10M of objects.
+
+**Warning:** Only add indexes for most used queries, an over indexed MongoDB will eat too much RAM.
+
+## Relational
+
+Using GraphQL do not change best practices to construct your database. In the **Queries** section we saw 2 types of relations:
```

Comment:
I'm not sure what the meaning of the first sentence is?

Reply:
"When using the GraphQL API, use the same best practices to construct your database."

Does that seem ok to you @Moumouls?

Reply:
I've gone ahead with my suggestion.